### PR TITLE
Move date of birth and ticket code check to venue config

### PR DESCRIPTION
**Pull request by thecadams**

### CODE REVIEW

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx [hunk 1]
```diff
@@ -171,36 +151,12 @@ const RegisterForm: React.FunctionComponent<PropsType> = ({
             <span className="input-error">Password is required</span>
           )}
         </div>
-        <TicketCodeField register={register} error={errors?.code} />
-        <DateOfBirthField register={register} error={errors?.date_of_birth} />
-        {IS_BURN &&
```
Comment (thecadams) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:176]: This came from removing the IS_BURN reference from this file; these hardcoded code of conduct questions won't be needed anymore.

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx [hunk 2]
```diff
@@ -98,15 +78,15 @@ const RegisterForm: React.FunctionComponent<PropsType> = ({
       afterUserIsLoggedIn && afterUserIsLoggedIn();
       closeAuthenticationModal();
       const accountProfileUrl = `/account/profile${
-        venueId ? `?venueId=${venueId}` : ""
+        venue.id ? `?venueId=${venue.id}` : ""
       }`;
-      history.push(IS_BURN ? "/enter/step2" : accountProfileUrl);
```
Comment (thecadams) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:103]: This came from removing the IS_BURN reference from this file.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx [hunk 3]
```diff
@@ -31,7 +29,9 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
 }) => {
   const firebase = useFirebase();
 
-  const history = useHistory();
+  const venue = useSelector(
+    (state) => state.firestore.ordered.currentVenue?.[0]
+  );
```
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:34]: I believe we already have a `const` selector for this pattern. Best to use that.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx [hunk 4]
```diff
@@ -44,9 +44,14 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
 
   const onSubmit = async (data: LoginFormData) => {
     try {
-      if (CODE_CHECK_ENABLED) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresEmailVerification)
+        await axios.get(codeCheckUrl(data.email));
```
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:49]: Do we need any of the return values from this? Or is it fine to just catch errors?

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx [hunk 5]
```diff
@@ -37,59 +35,41 @@ export interface RegisterData {
...
-    text:
-      "I will endeavor not to create indecent experiences or content, and understand my actions may be subject to review and possible disciplinary action",
-  },
-  {
-    name: "regionalBurn",
-    text:
-      "I understand this is an Australian Regional Burn guided by the Ten Principles of Burning Man",
-  },
-];
-
```
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:57]: As chris commented elsewhere, these hardcoded questions won't be required anymore

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx [hunk 6]
```diff
@@ -37,59 +35,41 @@ export interface RegisterData {
...
   displayLoginForm,
   displayPasswordResetForm,
   afterUserIsLoggedIn,
   closeAuthenticationModal,
 }) => {
   const history = useHistory();
-  const venue = useSelector((state) => state.firestore.ordered.currentVenue);
+  const venue = useSelector(
+    (state) => state.firestore.ordered.currentVenue?.[0]
+  );
```
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:47]: Use the `const` selector we have for this

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx [hunk 7]
```diff
@@ -37,59 +35,41 @@ export interface RegisterData {
...
-      if (CODE_CHECK_ENABLED) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
+      if (venue.requiresEmailVerification)
+        await axios.get(codeCheckUrl(data.email));
       const auth = await signUp(data);
-      if (CODE_CHECK_ENABLED && auth.user) {
+      if (
+        auth.user &&
+        (venue.requiresTicketCode || venue.requiresEmailVerification)
+      ) {
```
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:72]: [DRY] This seems the same as another block of similar code. Can DRY this up in future.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx (hunk 4 above)
Comment (thecadams) [reply src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:49]: Yep we just catch errors so all good

**Review by 0xdevalias — APPROVED** [review]
LGTM!

**Pull request merged**